src/libcprover-cpp/api.cpp

```diff
@@ -105,3 +115,47 @@ void api_sessiont::load_model_from_files(const std::vector<std::string> &files)
   implementation->model = util_make_unique<goto_modelt>(initialize_goto_model(
     files, *implementation->message_handler, *implementation->options));
 }
+
+void api_sessiont::verify_model()
```

Comment:
⛏️ 
In my opinion this function is a bit too long and complex to be in an API.

Can it be rewritten moving the logics into `goto-checker` or somewhere else?

Reply:
This reflects the complexity of the existing internal APIs. There isn't a whole lot we can do without refactoring the rest of the code base, for which would need buy-in to whatever we come up with from more code owners. If its any consolation, this is a lot easier to go through than `cbmc_parse_options.cpp`, which is the equivalent for the `cbmc` command line binary. A lot of what is in this function is also comments, explaining why the calls are needed in the order they are in.